### PR TITLE
Added nginx module

### PR DESCRIPTION
**Pull request by Nashluffy**
This adds support for the nginx ingress controller. Similar to the Traefik support, and includes an nginx ingress dashboard. 

The module assumes a few things about how nginx ingress is installed in your cluster...
- Prometheus metrics are being exported from the nginx controller on port `10254` which is the de-facto port. See [here](https://docs.gitlab.com/ee/user/project/integrations/prometheus_library/nginx_ingress.html#manually-setting-up-nginx-ingress-for-prometheus-monitoring) to expose metrics if installed via raw yaml, or [here](https://docs.gitlab.com/ee/user/project/integrations/prometheus_library/nginx_ingress.html#about-managed-nginx-ingress-deployments) if installed via helm chart.
- All related resources are installed in the `ingress-nginx` namespace (default if following suggested installation)
- The module creates a service and exposes the prometheus port with a known name, which removes chance of name mismatch. It's not gonna mess anything up to have it defined twice, but may cause confusion if you aren't aware.  

Here's a screenshot of what the dashboard looks like.

Screenshot from 2020-08-16 00-24-05

### CODE REVIEW

**Comment by Nashluffy** [comment]
Oops, I realize I accidentally committed a bunch of manifest files. Fixing now.

File: vars.jsonnet [hunk 1]
```diff
@@ -25,6 +25,11 @@
       enabled: false,
       file: import 'modules/metallb.jsonnet',
     },
+    {
+      name: 'nginxExporter',
+      enabled: true,
```
Comment (damacus) [inline vars.jsonnet:30]: Seeing we we don't install both Nginx & Traefik by default, should we default this to false?

Comment (Nashluffy) [reply vars.jsonnet]: Agreed, changed it to false.

**Comment by carlosedp** [comment]
This is very nice, thanks for the PR and the review!

**Pull request merged**